types/time.go

```diff
@@ -1820,10 +1827,115 @@ func ExtractTimeValue(unit string, format string) (int64, int64, int64, float64,
 	case "YEAR_MONTH":
 		return extractYearMonth(format)
 	default:
-		return 0, 0, 0, 0, errors.Errorf("invalid singel timeunit - %s", unit)
+		return 0, 0, 0, 0, errors.Errorf("invalid single timeunit - %s", unit)
 	}
 }
 
+// ExtractDurationValue extract the value from format to Duration.
+func ExtractDurationValue(unit string, format string) (Duration, error) {
+	switch strings.ToUpper(unit) {
```

Comment:
I mean here you can

```
unit = strings.ToUpper(unit)
switch unit {
 ...
}
```

Make sure that `MICROSEconD` `SeConD` all work likewise (case insensitive) in mysql

Reply:
yes it can work.